### PR TITLE
ENGG-2319: Add $1 support in file path in modify response rule

### PR DESCRIPTION
**Pull request by RuntimeTerror10**

### CODE REVIEW

File: common/rule-processor/src/processors/ModifyResponseRuleProcessor.js [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+    }
+    const resultingUrl = RuleHelper.matchUrlWithRulePair(pair, requestURL, details);
+    if (resultingUrl === null) return null;
 
     if (!pair.response || !pair.response.type || !pair.response.value) return null;
 
     return {
       action: "modify_response",
-      response: pair.response.value,
+      response: isResponseTypeLocalFile ? resultingUrl : pair.response.value,
```
Comment (nsrCodes) [inline common/rule-processor/src/processors/ModifyResponseRuleProcessor.js:19]: please write the tests for these

File: common/constants.js [hunk 2]
```diff
@@ -198,11 +198,6 @@ CONSTANTS.SCRIPT_LIBRARIES = {
   },
 };
 
-CONSTANTS.REQUEST_BODY_TYPES = {
-  STATIC: "static",
-  CODE: "code",
-};
-
```
Comment (nsrCodes) [inline common/constants.js:205]: why delete these?

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/ResponseBodyRow/index.js [hunk 3]
```diff
@@ -106,7 +109,64 @@ const ResponseBodyRow = ({ rowIndex, pair, pairIndex, ruleDetails, isInputDisabl
               >
                 {pair.response.value ? "Change file" : " Select file"}
               </RQButton>
-              <span className="file-path"> {pair.response.value ? pair.response.value : " No file chosen"}</span>{" "}
+              {pair.source.operator === GLOBAL_CONSTANTS.RULE_OPERATORS.MATCHES ||
```
Comment (nsrCodes) [inline app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/ResponseBodyRow/index.js:112]: why also for matches?

**Review by nsrCodes — CHANGES_REQUESTED** [review]

File: app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/ResponseBodyRow/index.js (hunk 3 above)
Comment (RuntimeTerror10) [reply app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/ResponseBodyRow/index.js:112]: We can capture groups from both regex and wildcard matches

File: common/constants.js (hunk 2 above)
Comment (RuntimeTerror10) [reply common/constants.js:205]: This is a duplicate, there's another definition of this constant just below this

**Review by nsrCodes — APPROVED** [review]

**Pull request merged**